forumsweats/commands/auction.py

```diff
@@ -0,0 +1,273 @@
+
+	description += f'\nReact with {AUCTION_EMOJI} to bid\n'
+
+	if ended:
+		if highest_bidder is None:
+			description = f'**No highest bidder**'
+		else:
+			description = f'**The highest bid was: {current_bid} by <@{highest_bidder}>**'
+	else:
+		ends_in_string = seconds_to_string(end - int(time.time()))
+		description += f'**Ends in: {ends_in_string}**'
+	description += f'\nHosted by: <@{creator_id}>'
+	
+
+	embed = discord.Embed(
+		title=item,
+		description=description,
+		timestamp=datetime.fromtimestamp(end)
+	)
+
+	return embed
+
+
+def handle_bids(message: Message, data):
+	@client.event
+	async def on_reaction_add(reaction: Reaction, user: User):
+		if user.bot or reaction.message.id != message.id: return
+		await reaction.remove(user)
+
+		auction_info = await db.get_auction(message.id)
+		highest_bid = auction_info["highest_bid"] or 0
+
+		# Check if auction ended
+		if auction_info["ended"] == True: return
+
+		# Check if the person has enough bobux
+		if not await db.get_bobux(user.id) > highest_bid + 100:
+			await user.send("You do not have enough bobux to bid")
+			return
+
+		await db.set_highest_bidder(message.id, highest_bid + 100, user.id)
+		data["highest_bid"] += 100
+
+		if data.get('end') - int(time.time()) < 30:
+			await db.extend_auction(message.id, 100)
+
+		embed = create_auction_embed(data, bidder=user.id)
+		await message.edit(embed=embed)
+
+async def create_new_auction(creator_id: int, channel: discord.abc.GuildChannel, length: int, item: str):
+	end = int(time.time() + length)
+
+	
+	embed = create_auction_embed({
+		'creator_id': creator_id,
+		'end': end,
+		'ended': False,
+		'item': item,
+	})
+
+	auction_message: discord.Message = await channel.send('💸 **Auction** 💸', embed=embed) # type: ignore (the typings on discord.py are wrong)
+	await auction_message.add_reaction(AUCTION_EMOJI)
+
+
+	auction_data = await db.create_new_auction(
+		message_id=auction_message.id,
+		creator_id=creator_id,
+		channel_id=channel.id, # type: ignore (the typings on discord.py are wrong)
+		end=end,
+		item=item,
+	)
+
+	handle_bids(auction_message, auction_data)
+
+	asyncio.ensure_future(continue_auction(auction_message.id), loop=client.loop)
+
+	return auction_message
+
+
+async def prompt_input(client: discord.Client, user: Member, channel: discord.abc.Messageable, prompt_message: str, invalid_message: str, check: Callable[[str], Any]) -> Any:
+	user_response = None
+
+	await channel.send(prompt_message)
+
+	while user_response is None:
+		m: discord.Message = await client.wait_for(
+			'message',
+			check=lambda m:
+				m.author.id == user.id
+				and channel.id == channel.id, # type: ignore (the typings on discord.py are wrong)
+			timeout=60
+		)
+
+		if m.content.lower() == 'cancel':
+			return await m.add_reaction('👍')
+		
+		user_response = await check(m.content)
+
+		if user_response is None:
+			await channel.send(invalid_message + ' (Type "cancel" to cancel the giveaway creation)')
```

Comment:
You forgot to change this to auction instead of giveaway

Reply:
Both fixed